caddyhttp/basicauth/basicauth.go

```diff
@@ -63,7 +63,7 @@ func (a BasicAuth) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error
 
 			// let upstream middleware (e.g. fastcgi and cgi) know about authenticated
 			// user; this replaces the request with a wrapped instance
-			r = r.WithContext(context.WithValue(r.Context(),
+			*r = *r.WithContext(context.WithValue(r.Context(),
```

Comment:
OK. However I think I need a bit of help, because I can't figure out what is going on.  

If I don't dereference `r`, then the context value is not available to the replacer.  Perhaps when someone has time they could try to figure out where the issue is.  I have even tried changing the context key just incase it was being overwritten.